### ISSUE TITLE
Request to add support for Tuxedo laptops

### ISSUE THREAD

**Issue by Raltay** [ublue-os/bluefin#898]
Hello,

I see that currently Bluefin has dedicated images for Framework/Surface/Asus laptops (as seen in Image Picker). Would it be possible/feasible to also add dedicated images for Tuxedo latpops?

From what I gather, this would require building/including kernel modules/drivers from [tuxedo-drivers](https://gitlab.com/tuxedocomputers/development/packages/tuxedo-drivers) and also preinstalling [Tuxedo Control Center](https://github.com/tuxedocomputers/tuxedo-control-center) app - which itself depends on said tuxedo-drivers.

I found that there was already ticket related to that in https://github.com/ublue-os/akmods/issues/30, but I'm not sure if any decision regarding it has been made. Also (in the meantime), tuxedo-keyboard has been deprecated in favor of tuxedo-drivers.

I'm asking this because I have a Tuxedo laptop on the way (Pulse 14 Gen 3) and there are some HW related features (KB backlight, fan control etc.) which won't work properly with Linux kernel without those drivers. If possible, I'd like to use Bazzite or Bluefin on this laptop since Bazzite has been great on my Steam Deck OLED so far.

Thank you in advance.

Best Regards

**Comment by castrojo** [comment]
Yeah this shouldn't be a problem, when those akmods get added we can just enable it in bazzite/bluefin. That'll also enable people to add support in their custom images.

**Comment by tulilirockz** [comment]
This cannot be implemented w/o automatically tainting the kernel since the kernel people got mad with the Tuxedo people :(

**Comment by salim-b** [comment]
> This cannot be implemented w/o automatically tainting the kernel since the kernel people got mad with the Tuxedo people :(

I think Tuxedo managed to avoid the changes that threatened to taint their out-of-tree kernel modules as proprietary by swiftly relicensing them to `GPL-2.0-or-later`. v4.11.3+ of `tuxedo-drivers` is 100% `GPL-2.0-or-later` [according to the license file](https://gitlab.com/tuxedocomputers/development/packages/tuxedo-drivers/-/blob/v4.11.3/debian/copyright?ref_type=tags). See also their [follow-up blog post](https://www.tuxedocomputers.com/en/License-Adjustment-and-Mainline-Integration-of-TUXEDO-Kernel-Drivers.tuxedo).

In short: [Pressure from upstream](https://www.phoronix.com/news/TUXEDO-Drivers-Taint-Patches) worked miracles. 😄

**Comment by BrickMan240** [comment]
Okay, I seemed to have missed this particular issue thread.

If anyone is interested, I have unofficial images for using Tuxedo drivers on Universal Blue. Aurora, Bluefin, Bazzite, all there. It's not the nicest implementation I'm sure, but it all seems to work so far, so feel free to use them as a temporary solution. You'd be be getting your hardware working before official support, and helping test the drivers for upstream implementation. Here's my repo:

[ublue-tuxedo-tcc](https://github.com/BrickMan240/ublue-tuxedo-tcc)

**Comment by castrojo** [comment]
No one's driving this, closing.
